modules/nw-exposing-router-metrics.adoc

```diff
@@ -0,0 +1,158 @@
+// Module included in the following assemblies:
+//
+// * networking/ingress_operator.adoc
+
+:_content-type: PROCEDURE
+[id="nw-exposing-router-metrics_{context}"]
+= Exposing router metrics
+
+You can expose the HAProxy router metrics by default in Prometheus format on the default stats port, 1936. The external metrics collection and aggregation systems such as Prometheus and statsd can access the HAProxy router metrics. You can view the HAProxy router metrics in a browser in the HTML and cluster service version (CSV) format.
+
+.Prerequisites
+
+* You configured your firewall to access the default stats port, 1936.
+
+.Procedure
+
+. A router pod creates a default username and password for the router. A router pod stores username and password at `/var/lib/haproxy/conf/metrics-auth/statsUsername` and `/var/lib/haproxy/conf/metrics-auth/statsPassword` respectively.
+
+.. Get the username by running the following command:
++
+[source,terminal]
+----
+$ oc rsh <router_pod> cat metrics-auth/statsUsername
+----
++
+.Example command
+[source,terminal]
+----
+$ oc rsh router-default-d78cffbc6-jcgfn cat metrics-auth/statsUsername
+----
++
+.Example output
+[source,terminal]
+----
+dXNlcmN3NnY2
+----
+
+.. Get the password by running the following command:
++
+[source,terminal]
+----
+$ oc rsh <router_pod> cat metrics-auth/statsPassword
+----
++
+.Example command
+[source,terminal]
+----
+$ oc rsh router-default-d78cffbc6-jcgfn cat metrics-auth/statsPassword
+----
++
+.Example output
+[source,terminal]
+----
+cGFzczdqbnJt
+----
+
+. Get the raw statistics in Prometheus format by running the following command:
++
+[source,terminal]
+----
+$ curl <user>:<password>@<router_IP>:<stats_port>
+----
++
+.Example command
+[source,terminal]
+----
+$ curl dXNlcmN3NnY2:cGFzczdqbnJt@10.254.254.35:1936/metrics
+----
+
+. Access the metrics securely by running the following command:
++
+[source,terminal]
+----
+curl -u user:password https://<router_IP>:<stats_port>/metrics -k
+----
++
+.Example command
+[source,terminal]
+----
+$ curl -u dXNlcmN3NnY2:cGFzczdqbnJt https://10.254.254.35:1936/metrics -k
+----
+
+. Access the default stats port, 1936, by running the following command:
++
+[source,terminal]
+----
+$ curl <user>:<password>@<router_IP>:<stats_port>
```

Comment:
please add `-u`  before <user>:<password>

Reply:
if add -u, "@" need be removed.
`sh-4.4$ curl -u dXNlcmN3NnY2:cGFzczdqbnJt 10.254.254.35:1936/metrics
^C
sh-4.4$ curl -u dXNlcmN3NnY2:cGFzczdqbnJt@10.254.254.35:1936/metrics
curl: no URL specified!
curl: try 'curl --help' or 'curl --manual' for more information
sh-4.4$ `

Reply:
yes, correct. Please follow below style and remove `@`
```
curl -u <user>:<password> http://<router_IP>:<stats_port>/metrics
```

Thanks for the update.

Reply:
@lihongan thanks for the review. 
@xenolinux  curl -u is more readable, please replace 

`curl <user>:<password>@<router_IP>:<stats_port>`   with :

`curl -u <user>:<password> http://<router_IP>:<stats_port>`

thanks.

Reply:
@lihongan @ShudiLi Updated all the `curl` instances with the `-u` option. 
One question: Will this url change too? `http://<user>:<password>@<router_IP>:<stats_port>`

Reply:
If using browser to open the url, please keep it as  `http://<user>:<password>@<router_IP>:<stats_port>`.
For curl command, please follow `curl -u <user>:<password> http://<router_IP>:<stats_port>`

Reply:
Addressed @lihongan 

Thanks